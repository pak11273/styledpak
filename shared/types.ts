import {
  AlignItemsProps,
  AlignSelfProps,
  BorderRadiusProps,
  ButtonStyleProps,
  ColorProps,
  DisplayProps,
  FlexboxProps,
  FontSizeProps,
  GridProps,
  HeightProps,
  LayoutProps,
  LineHeightProps,
  MaxHeightProps,
  MaxWidthProps,
  MinHeightProps,
  MinWidthProps,
  OrderProps,
  OverflowProps,
  SpaceProps,
  TypographyProps,
  WidthProps,
} from "styled-system"

export type SystemProps = AlignSelfProps &
  AlignItemsProps &
  AlignSelfProps &
  ButtonStyleProps &
  BorderRadiusProps &
  ColorProps &
  DisplayProps &
  FlexboxProps &
  FontSizeProps &
  GridProps &
  HeightProps &
  LayoutProps &
  LineHeightProps &
  MaxHeightProps &
  MaxWidthProps &
  MinHeightProps &
  MinWidthProps &
  OrderProps &
  OverflowProps &
  SpaceProps &
  TypographyProps &
  WidthProps
