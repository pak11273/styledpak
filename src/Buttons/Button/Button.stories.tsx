import { boolean, color, object, select, text } from "@storybook/addon-knobs"

import Button from "./Button"
import React from "react"
import { storiesOf } from "@storybook/react"

const onClick = () => console.log("this is a button")

storiesOf("Button", module)
  .add("no props", () => (
    <div style={{ display: "flex", justifyContent: "center" }}>
      <Button to="#">No Props</Button>
    </div>
  ))
  .add("with icon", () => (
    <div style={{ display: "flex", justifyContent: "center" }}>
      <Button
        to="#"
        focusColor={text("focusColor", "red")}
        padding={text("padding", "6px 16px")}
        margin={text("margin", "0px")}
        shadow={boolean("shadow", true)}
        onClick={onClick}
      >
        <div>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            width="24"
            height="24"
            fill="currentcolor"
          >
            <circle
              r={11}
              cx={12}
              cy={12}
              fill="none"
              stroke="currentcolor"
              strokeWidth={2}
            />
          </svg>
        </div>
      </Button>
    </div>
  ))
  .add("playground", () => (
    <div style={{ display: "flex", justifyContent: "center" }}>
      <Button
        bg={text("bg", null)}
        padding={text("padding", "6px 16px")}
        margin={text("margin", "0px")}
        shadow={boolean("shadow", true)}
        // disabled={boolean("disabled", false)}
        variant={text("variant", null)}
        onClick={onClick}
      >
        <div>Custom</div>
      </Button>
    </div>
  ))
