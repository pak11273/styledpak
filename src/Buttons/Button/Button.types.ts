// Generated with util/create-component.js
export interface ButtonProps {
  background?: string
  border?: string
  case?: string
  color?: string
  cursor?: string
  focus?: string
  padding?: string
  margin?: string
  shadow?: boolean
  theme?: object
  lowercase?: boolean
  info?: boolean
  radius?: string
  style?: React.CSSProperties
}
