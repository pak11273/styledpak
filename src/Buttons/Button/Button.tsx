import React, { ReactElement, useEffect, useRef } from "react"

import { ButtonProps } from "./Button.types"
import styled from "styled-components"

const Style = styled.button<ButtonProps>`
  background: ${props => props.bg || props.theme.colors.pak.default};
  border: ${props =>
    props.color
      ? `1px solid ${props.color}`
      : props.border
      ? props.border
      : "none"};
  border-radius: ${props => props.radius || "3px"};
  color: ${props => props.color};
  cursor: ${props => props.cursor || "pointer"};
  &:focus {
    background: ${props => props.focusColor || "none"};
  }
  margin: ${props => props.margin || "0"};
  outline: none;
  padding: ${props => props.padding || "6px 16px"};
  text-transform: ${props => props.case || "uppercase"};
  box-shadow: ${props =>
    props.shadow
      ? "0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)"
      : "none"};
`

const Button: React.FC<ButtonProps> = props => {
  const btn = useRef(null)
  // TODO: when not focused change bg back to default
  const handleFocusOut = (e: MouseEvent) => {
    console.log("btn: ", btn)
  }
  useEffect(() => {
    addEventListener("mouseleave", handleFocusOut)
    return removeEventListener("mouseleave", () => {})
  })
  return <Style {...props} onClick={props.onClick} ref={btn} />
}

export default Button
