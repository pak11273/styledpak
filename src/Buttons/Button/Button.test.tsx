// import "@testing-library/jest-dom/extend-expect"

import Button from "./Button"
import { ButtonProps } from "./Button.types"
// Generated with util/create-component.js
import React from "react"
import { render } from "@testing-library/react"

describe("<Button />", () => {
  let props: ButtonProps

  const renderComponent = () => render(<Button {...props} />)

  it("should render as a default button", () => {
    expect(renderComponent).toMatchSnapshot()
  })
})
