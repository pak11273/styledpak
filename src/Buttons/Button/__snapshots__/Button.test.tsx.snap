// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render as a default button 1`] = `[Function]`;
